SimTKcommon/Simulation/include/SimTKcommon/internal/MeasureImplementation.h

```diff
@@ -1289,6 +1289,43 @@ template <class T>
 class Measure_Differentiate_Result {
 public:
     Measure_Differentiate_Result() : derivIsGood(false) {}
+
+    Xml::Element toXmlElement(const std::string& name) const {
+        static const int version = 1;
+        Xml::Element e("Measure_Differentiate_Result");
+        if (!name.empty()) e.setAttributeValue("name", name);
+        e.setAttributeValue("version", String(version));
+        e.appendNode(toXmlElementHelper(operand, "operand", true));
+        e.appendNode(toXmlElementHelper(operandDot, "operandDot", true));
+        e.appendNode(toXmlElementHelper(derivIsGood, "derivIsGood", true));
```

Comment:
@klshrinidhi and I just had a whiteboard session and he brought up two other points:

1. Maybe it is desirable to prefer a free function (that is, template specialization of `toXmlElement()`) over a member function) if both exist. Why would one define the free function if the class already had `toXmlElement()`/`fromXmlElement()`? Perhaps the only reason would be because they want to override the behavior of the member function to achieve something new. Probable answer: the control of serializing a type should be held by that type; it would be dangerous for others to subsume that control. The resulting Xml file could not be loaded correctly by others who did not have that free function but were using the same class.

2. Drawing a parallel to `std::begin()` and `std::end()`, @klshrinidhi was saying how you can call `begin()` or `end()` (without any namespace qualification), and if the argument is in the `std` namespace then `std::begin()`/`std::end()` will be called. He raised the idea that it could be nice to never have to qualify `toXmlElement()` (or `toXmlElementHelper()`) with `SimTK::`. This would require a `<ns>::toXmlElement()` in each namespace that wanted to use it, though.